packages/api/es/indexer.js

```diff
@@ -344,30 +387,39 @@ async function reingest(g) {
  * @returns {Promise} object with response from the three indexer
  */
 async function handlePayload(event) {
-  let payload;
-  const source = get(event, 'EventSource');
+  const executionModel = new Execution();
+  const granuleModel = new Granule();
+  const pdrModel = new Pdr();
 
-  if (source === 'aws:sns') {
-    payload = get(event, 'Sns.Message');
-    payload = JSON.parse(payload);
-  }
-  else {
-    payload = event;
-  }
+  const payload = event.EventSource === 'aws:sns'
+    ? JSON.parse(event.Sns.Message)
+    : event;
+
+  const executionRecord = await buildExecutionRecord(executionModel, payload);
+  const pdrRecord = buildPdrRecord(pdrModel, payload);
+  const granuleRecords = await buildGranuleRecords(granuleModel, payload);
 
-  let executionPromise;
-  const e = new Execution();
-  if (['failed', 'completed'].includes(payload.meta.status)) {
-    executionPromise = e.updateExecutionFromSns(payload);
+  const recordsCount = flatten([executionRecord, granuleRecords, pdrRecord]).length;
+
+  if (recordsCount > 10 || inTestMode()) {
+    if (executionRecord) await executionModel.create(executionRecord);
+    if (pdrRecord) await pdrModel.create(pdrRecord);
+    await granuleModel.create(granuleRecords);
   }
-  else {
-    executionPromise = e.createExecutionFromSns(payload);
+  else if (recordsCount > 0) {
```

Comment:
Is this `(recordsCount > 0)` always true?  If not, should we check it earlier (before `if (recordsCount > 10 || inTestMode())`)?

Reply:
Do we have tests to verify that the transaction actually works?  e.g. If one of the statement fails, the whole transaction is not executed/rolled back?

Reply:
> Is this `(recordsCount > 0)` always true? If not, should we check it earlier (before `if (recordsCount > 10 || inTestMode())`)?

Changed in 54e911395